### PR TITLE
Improve code quality and unit test coverage of the Azure extension

### PR DESCRIPTION
**Pull request by zachjsh**
### Description

* Update unit tests to increase test coverage for the extension
* Clean up any messy code
* Enfore code coverage as part of tests.

<hr>

This PR has:
- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

<hr>

### CODE REVIEW

File: extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentPusher.java [hunk 1]
```diff
@@ -157,6 +128,7 @@ public DataSegment push(final File indexFilesDir, final DataSegment segment, fin
       );
     }
     catch (Exception e) {
+      // do we want to throw a specfic exception here? Interface only expects IOExceptions to be thrown.
```
Comment (jihoonson) [inline extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentPusher.java]: Haha, is this question for the reviewers? I think `RuntimeException` is enough for now.

File: extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java [hunk 2]
```diff
@@ -95,13 +87,34 @@ public long getBlobLength(final String containerName, final String blobPath)
...
     return getCloudBlobContainer(container).getBlockBlobReference(blobPath).exists();
   }
+
+  @VisibleForTesting
+  CloudBlobClient getCloudBlobClient()
+  {
+    return this.cloudBlobClient;
+  }
+
+  private CloudBlobContainer getCloudBlobContainer(final String containerName)
```
Comment (jihoonson) [inline extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java]: nit: this method not just returns a container, but can create it if it doesn't exist. Maybe better to rename to something more clear such as `getOrCreateCloudBlobContainer()`.

Comment (zachjsh) [reply extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java]: fixed

File: extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentPusher.java (hunk 1 above)
Comment (zachjsh) [reply extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentPusher.java]: removed comment

File: extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentPusher.java [hunk 3]
```diff
@@ -179,4 +151,36 @@ public DataSegment push(final File indexFilesDir, final DataSegment segment, fin
...
+      throws StorageException, IOException, URISyntaxException
+  {
+    azureStorage.uploadBlob(compressedSegmentData, config.getContainer(), azurePath);
+
+    final DataSegment outSegment = segment
+        .withSize(size)
+        .withLoadSpec(this.makeLoadSpec(new URI(azurePath)))
+        .withBinaryVersion(binaryVersion);
+
+    log.info("Deleting file [%s]", compressedSegmentData);
```
[the remark this answers is not in the document]
Comment (zachjsh) [reply extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentPusher.java]: fixed

**Pull request merged**